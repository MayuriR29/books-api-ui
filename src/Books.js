import React, { Component } from "react";
import { API_HOST } from "./config";

class Books extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
   async componentDidMount() {
     await this.getBooks();
  }
  
  async getBooks() {
    const url = `${API_HOST}/books`;
    const response = await fetch(url);
    const books = await response.json();
    this.setState({
      books: books
    });
  }
  render() {
    return (
      <div>
        <h1>Books List</h1>
        {this.state.books.map(book => {
          return <li key={book._id}>{book.title}</li>;
        })}
      </div>
    );
  }
}
export default Books;
