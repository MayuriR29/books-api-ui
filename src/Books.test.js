import React from "react";
import Books from "./Books";
global.fetch = require('jest-fetch-mock');
import ShallowRenderer from "react-test-renderer/shallow";

it("test title is returned when render is called first ", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Books />);
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("test title and booklist after component mounted ", async() => {
  fetch.mockResponseOnce(JSON.stringify([
      {
        title:"Harry Potter",
        _id:"wfwef23244"
      },
      {
        title:"Jungle Book",
        _id:"asdqd324"
      }
  ]));
 const renderer = new ShallowRenderer();
  renderer.render(<Books />);

  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
