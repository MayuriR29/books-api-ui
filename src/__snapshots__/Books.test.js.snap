// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test title and booklist after component mounted  1`] = `
<div>
  <h1>
    Books List
  </h1>
  <li>
    Harry Potter
  </li>
  <li>
    Jungle Book
  </li>
</div>
`;

exports[`test title is returned when render is called first  1`] = `
<div>
  <h1>
    Books List
  </h1>
</div>
`;
